Updates to KES docs for config reference (#48)
minio/kes-docs: content/integrations/aws-secrets-manager.md

 
2. Generate Client Credentials

>    Create credentials the client application uses to access the KES server. 

Reviewer: maybe
```suggestion
   Create credentials to allow the client application to access the KES server. 
```
but also this sentence feels a bit redundant? The heading just above does say "Generate Client Credentials"

Author: You're right, it's redundant. ✂️

     ciphertext: eyJhZWFkIjoiQUVTLTI1Ni1HQ00tSE1BQy1TSEEtMjU2IiwiaWQiOiIxMTc1ZjJjNDMyMjNjNjNmNjY1MDk5ZDExNmU3Yzc4NCIsIml2IjoiVHBtbHpWTDh5a2t4VVREV1RSTU5Tdz09Iiwibm9uY2UiOiJkeGl0R3A3bFB6S21rTE5HIiwiYnl0ZXMiOiJaaWdobEZrTUFuVVBWSG0wZDhSYUNBY3pnRWRsQzJqWFhCK1YxaWl2MXdnYjhBRytuTWx0Y3BGK0RtV1VoNkZaIn0=
   }
   ```

## Using KES with a MinIO Server

MinIO Server requires KES to enable server-side data encryption.

See the [KES for MinIO instruction guide]({{< relref "/tutorials/kes-for-minio.md" >}}) for additional steps needed to use your new KES Server with a MinIO Server.

## Configuration References

> The following section describes the Key Encryption Service (KES) configuration settings to use AWS Secrets Manager and AWS Key Management System as the root KMS for Server Side Encryption.

Reviewer 2: So this is a ridiculous nuance, but strictly speaking KES doesn't perform SSE.

It just uses the root KMS for storage and management of External Keys, which are then used to produce cryptographic keys for use with *anything*. 

MinIO happens to use those keys for SSE, but there's no reason any other client could use it for anything else.

Author: replacing `for Server Side Encryption` with `to store external keys, such as the keys used for Server-Side Encryption on a MinIO Server` throughout.

| `policy`                      | Specify one or more [policies]({{< relref "/tutorials/configuration.md#policy-configuration" >}}) to control access to the KES server. MinIO SSE requires access to the following KES cryptographic APIs: <br><br> `/v1/key/create/*` <br> `/v1/key/generate/*` <br> `/v1/key/decrypt/*` <br><br> Specifying additional keys does not expand MinIO SSE functionality and may violate security best practices around providing unnecessary client access to cryptographic key operations. <br><br> You can restrict the range of key names MinIO can create as part of performing SSE by specifying a prefix before the `*.` For example, `minio-sse-*` only grants access to `create`, `generate`, or `decrypt` keys using the `minio-sse-` prefix. <br><br>KES uses mTLS to authorize connecting clients by comparing the hash of the TLS certificate against the `identities` of each configured policy. Use the `kes identity of` command to compute the identity of the MinIO mTLS certificate and add it to the `policy.<NAME>.identities` array to associate MinIO to the `<NAME>` policy. |

> | `keys`                        | Specify an array of keys which *must* exist on the root KMS for KES to successfully start. KES attempts to create the keys if they do not exist and exits with an error if it fails to create one or more key. KES does not accept any client requests until it completes validation of all specified keys.|

Reviewer: ```suggestion
| `keys`                        | Specify an array of keys which *must* exist on the root KMS for KES to successfully start. KES attempts to create the keys if they do not exist and exits with an error if it fails to create one or more keys. KES does not accept any client requests until it completes validation of all specified keys.|
```
likely the other locations as well, yay automated find/replace

Author: Nice catch. Fixed throughout.

     ciphertext: eyJhZWFkIjoiQUVTLTI1Ni1HQ00tSE1BQy1TSEEtMjU2IiwiaWQiOiIxMTc1ZjJjNDMyMjNjNjNmNjY1MDk5ZDExNmU3Yzc4NCIsIml2IjoiVHBtbHpWTDh5a2t4VVREV1RSTU5Tdz09Iiwibm9uY2UiOiJkeGl0R3A3bFB6S21rTE5HIiwiYnl0ZXMiOiJaaWdobEZrTUFuVVBWSG0wZDhSYUNBY3pnRWRsQzJqWFhCK1YxaWl2MXdnYjhBRytuTWx0Y3BGK0RtV1VoNkZaIn0=
   }
   ```

## Using KES with a MinIO Server

> MinIO Server requires KES to set up server-side data encryption.

Reviewer: `set up` -> `enable` perhaps? It isn't a thing you do once then it's over.

See the [KES for MinIO instruction guide]({{< relref "/tutorials/kes-for-minio.md" >}}) for additional steps needed to use your new KES Server with a MinIO Server.

## Configuration References

> The following section describes each of the Key Encryption Service (KES) configuration settings for using AWS Secrets Manager and AWS Key Management System as the root KMS for Server Side Encryption with KES.

Reviewer: would this shorter alternative still be accurate?
```suggestion
The following section describes the Key Encryption Service (KES) configuration settings to use AWS Secrets Manager and AWS Key Management System as the root KMS for Server Side Encryption.
```

```yaml {.copy}
address: 0.0.0.0:7373
root: ${ROOT_IDENTITY}

tls:
  key: kes-server.key
  cert: kes-server.cert

api:
  /v1/ready:
    skip_auth: false
    timeout:   15s

policy:
  minio-server:
    allow:
    - /v1/key/create/*
    - /v1/key/generate/*
    - /v1/key/decrypt/*
    - /v1/key/bulk/decrypt
    - /v1/key/list/*
    - /v1/status
    - /v1/metrics
    - /v1/log/audit
    - /v1/log/error
    deny:
    - /v1/key/generate/my-app-internal*
    - /v1/key/decrypt/my-app-internal*
    identities:
    - ${MINIO_IDENTITY}

    my-app:
    allow:
    - /v1/key/create/my-app*
    - /v1/key/generate/my-app*
    - /v1/key/decrypt/my-app*
    deny:
    - /v1/key/generate/my-app-internal*
    - /v1/key/decrypt/my-app-internal*
    identities:
    - df7281ca3fed4ef7d06297eb7cb9d590a4edc863b4425f4762bb2afaebfd3258
    - c0ecd5962eaf937422268b80a93dde4786dc9783fb2480ddea0f3e5fe471a731

keys:
  - name: "minio-encryption-key-alpha"
  - name: "minio-encryption-key-baker"
  - name: "minio-encryption-key-charlie"

cache:
  expiry:
    any: 5m0s
    unused: 20s
    offline: 0s

# The following log configuration only affects logging to console.
log:

>   # Enable/Disable logging error events to STDERR. Valid values
>   # are "on" or "off". If not set the default is "on". If no error
>   # events should be logged to STDERR it has to be set explicitly
>   # to: "off".

Reviewer: A little more direct alternative?
```suggestion
  # Enable/Disable logging error events to STDERR. 
  # Valid values are "on" or "off". Defaults to "on".
  # To prevent logging error events to STDERR, set to "off".
```

There's some verbose comments in this sample configuration, might be worth looking at others as well.

Author: This was just copy/paste. I will simplify throughout.

Author: Especially because I have the reference tab, so the comments are needed in the yaml.

Reviewer: How about after that I make a PR to improve the source, then yours will match 😁

| `policy`                      | Specify one or more [policies]({{< relref "/tutorials/configuration.md#policy-configuration" >}}) to control access to the KES server. MinIO SSE requires access to the following KES cryptographic APIs: <br><br> `/v1/key/create/*` <br> `/v1/key/generate/*` <br> `/v1/key/decrypt/*` <br><br> Specifying additional keys does not expand MinIO SSE functionality and may violate security best practices around providing unnecessary client access to cryptographic key operations. <br><br> You can restrict the range of key names MinIO can create as part of performing SSE by specifying a prefix before the `*.` For example, `minio-sse-*` only grants access to `create`, `generate`, or `decrypt` keys using the `minio-sse-` prefix. <br><br>KES uses mTLS to authorize connecting clients by comparing the hash of the TLS certificate against the `identities` of each configured policy. Use the `kes identity of` command to compute the identity of the MinIO mTLS certificate and add it to the `policy.<NAME>.identities` array to associate MinIO to the `<NAME>` policy. |

> | `keys`                        | Specify an array of keys which *must* exist on the root KMS for KES to successfully start. KES attempts to create the keys if they do not exist and exits with an error if it fails to create any key. KES does not accept any client requests until it completes validation of all specified keys.|

Reviewer: Would `fails to create any key` mean `fails to create one or more required keys`?

Author: one or more. FIxed throughout.

| `cache`                       | Specify expiration of cached keys in `#d#h#m#s` format. Unexpired keys may be used in the event the KMS becomes temporarily unavailable. <br><br> Entries may be set for `any` key, `unused` keys, or `offline` keys. <br><br> If not set, KES uses values of `5m` for all keys, `20s` for unused keys, and `0s` for offline keys. |
| `log`                         | Enable or disable output for `error` and `audit` type logging events to the console. |
| `keystore.aws.secretsmanager` | Use this section of values to configure the AWS Secrets Manager and AWS KMS. |

> | `keystore.aws.secretsmanager.endpoint` | The endpoint for the Secrets Manager service, including the region. |

Reviewer: "endpoint including region" suggests an example might be helpful? Is there something that makes sense and isn't too long?

Author: Added!

| `keystore.aws.secretsmanager.region` | The AWS region to use for other AWS services. |
| `keystore.aws.secretsmanager.kmskey` | The root KMS Key to use for cryptographic operations. Formerly known as the Customer Master Key. |

> | `keystore.aws.secretsmanager.credentials` | The AWS Credentials to use for performing authenticated operations against Secrets Manager and KMS. The specified credentials *must* have the [appropriate permissions](https://min.io/docs/minio/container/operations/server-side-encryption/configure-minio-kes-aws.html#minio-sse-aws-prereq-aws?ref=kes-docs). <br><br> Make entries for both `accesskey` and `secretkey` or an entry for `token`. |

Reviewer: Does this mean either "accesskey and secretkey" or "token"? 
> Make entries for both `accesskey` and `secretkey` or an entry for `token`.

Is this a reasonable interpretation?
"Create entries for either an accesskey/secretkey pair, or a token."

Author: reworked
